doc/CONTRIBUTING.md

```diff
@@ -1,59 +1,108 @@
 
-To run the full suite of unit tests, invoke the `run-unittests` script at
-the top-level of the pbench repository.
+1. It is a good practice to assign the issue to yourself to let others know you're working upon it.
 
-# Python formatting
+![Image](https://lh6.googleusercontent.com/MsF7-9lTnDgQsi888PqSFtWCOfoxnhDQVfNv6aem7AcEo06gDrCk-ISZ8C3VIe6AUfRfglRq3xgaHSzS2yfQHutWkbHingQswCYGpBqVOaa_WImECxYvKAZM-i6lr7HirFDAhA41)
 
-This project uses the [flake8](http://flake8.pycqa.org/en/latest) method of
-code style enforcement, linting, and checking.
+## Making changes to the codebase
 
-All python code contributed to pbench must match the style requirements. These
-requirements are enforced by the [pre-commit](https://pre-commit.com) hook
-using the [black](https://github.com/psf/black) Python code formatter.
+- Follow the instructions in the README.md to setup and install pbench.
+- Save your changes by creating your own local branches on git
 
-## Use pre-commit to set automatic commit requirements
+## Add, Commit and Push
+
+- Follow these commands to push the changes to your branch.
+```
+git add .
+git commit -m "Issue solved"
+git push origin branch_name
+```
 
-This project makes use of [pre-commit](https://pre-commit.com/) to do automatic
-lint and style checking on every commit containing Python files.
+## Conventions on commits, PRs, and overall git best practices.
 
-To install the pre-commit hook, run the executable from your Python 3 framework
-while in your current pbench git checkout:
+- Commit messages should have a short description (50 - 70 characters) followed by a longer format description of the changes below if needed. You’ll also notice each line is formatted for a specific length in the longer format description. For example:
+
+```
+Extend auditing to incoming, results, and users
+
+The server audit is now applied to the incoming, results, and users
+directory hierarchies.  Any unpacked tar ball should now be compre-
+hensively checked to see that all is in the correct place.
+
+The test-20 unit test gold file holds an example of an audit report
+covering all the possible outputs it can emit.  Each unit test runs
+the report as well, and they have been updated accordingly.
+```
 
+- For more on best practices, check out this article for reference from time to time: https://www.git-tower.com/learn/git/ebook/en/command-line/appendix/best-practices
+
+## Opening a pull request
+1. **If there are multiple commits, squash down the commits to one**
+  1. For more complicated commits it is appropriate to have more than one
+1. Commit the changes
+1. Click on New Pull Request
+1. Write appropriate Pull Request Title stating the fix
+  1. Use present tense (ex. Fixes, Changes, Fixing, Changing..)
+1. Reference the issue that the PR is fixing with **“Fixes #issue_number”** in the description
+1. Provide a detailed description (at least 50 - 70 characters) of the changes; if UI related, add screenshots or gif
```

Comment:
"(at least 50 - 70 characters)" seems out of place. Also "gifs" or "gif images"? Are only GIFS allowed?